script.R

```diff
@@ -10,6 +10,53 @@ if_error <- function(x, y = NULL) {
 `%||%` <- function(x, y) {
   if (!length(x) || is.null(x)) y else x
 }
+check_if_pkg_available <- function(pkg, ver = NULL) {
+  if (is.null(ver)) {
+    length(available.packages(filter = list(add = TRUE, function(db) db[db[, "Package"] == pkg, ]))) > 0
+  } else {
+    length(
+      available.packages(
+        filter = list(
+          add = TRUE,
+          function(db) {
+            db[db[, "Package"] == pkg & db[, "Version"] == ver, ]
+          }
+        )
+      )
+    ) > 0
+  }
+}
+add_to_minicran <- function(x, minicran_path) {
+  temp_dir <- tempfile()
+  on.exit(unlink(temp_dir))
+  dir.create(temp_dir)
+  new_file_name <- gsub("(.*?_.*)_.*?(\\..*)", "\\1\\2", basename(x))
```

Comment:
ugh, can we get at least a small comment on what's going on in this regex :P
Does it clean all `.` and `_` signs? Maybe two fixed `gsub` would be easier to understand.

This is what regex translator says about this regex

<img width="668" alt="image" src="https://github.com/user-attachments/assets/a7b52e73-95d1-4f79-83d9-be9984b977e4">

Reply:
```suggestion
  # for GH packages we have `pkg_1.2.3.9000_hash.tar.gz` so we need to remove the hash and keep only `pkg_1.2.3.9000.tar.gz`
  new_file_name <- gsub("(.*?_.*)_.*?(\\..*)", "\\1\\2", basename(x))
```